### ISSUE TITLE
Make C++ library installable

### ISSUE THREAD

**Issue by nachovizzo** [PRBonn/kiss-icp#330]
I knew this might happen sooner or later. I think the underlying problem is not the name of the targets but the lack of an installable C++ library. 

I postponed this always, but I'd say we could now fix this the proper way. Using modern cmake, and relying on exported targets instead of manually adding directories in the current build space

A probably outdated example:
https://github.com/PRBonn/vdbfusion/blob/main/src/vdbfusion/vdbfusion/cmake/Install.cmake

 In summary,  I think we should make the  C++ library installable, like is the case with all the dependencies we use

_Originally posted by @nachovizzo in https://github.com/PRBonn/kiss-icp/issues/327#issuecomment-2061075797_

**Comment by tizianoGuadagnino** [comment]
I will close this for my mental health, there are many other things moving for the project and I couldn't find the time to sit and just do it. I will close it for now, as we are gonna still be busy for sometimes but @nachovizzo @benemer you got my word I will come back to this.
